### PR TITLE
highlight winning cells

### PR DESCRIPTION
**Pull request by GrandSuccess87**
add methods to retrieve winning indices in the mock
update test suite
update view layers
update css

### CODE REVIEW

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js [hunk 1]
```diff
@@ -51,9 +43,34 @@ class ComputerVsHumanGameComponent extends Component {
...
+        this.setState({
+          gameStatus: Statuses.COMPUTER_THINKING
+        });
+        setTimeout( () => {this.computerMove();}, 1000);
+      }
+
+      if (this.state.gameStatus !== Statuses.IN_PROGRESS || this.state.gameStatus !== Statuses.COMPUTER_THINKING) {
+        this.highlightWinners(marks);
+      }
+    }
```
Comment (soft-circles) [inline src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:68]: There's a lot happening in `updateBoardState`. Can we extract these into three different methods?

File: src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js [hunk 2]
```diff
@@ -30,6 +30,20 @@ toggleMarker = async (event) => {
...
+
+  if (this.state.gameStatus !== Statuses.IN_PROGRESS) {
+    this.highlightWinners(marks);
+  }
+}
+
+highlightWinners = async (marks) => {
+  let data = await PhoenixApi.getWinningIndices(marks);
+  let winningCells = data.index_list;
+  console.log(winningCells);
```
Comment (soft-circles) [inline src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js]: remove this `console.log`

File: src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js [hunk 3]
```diff
@@ -30,6 +30,20 @@ toggleMarker = async (event) => {
...
     gameStatus: await PhoenixApi.getStatus(marks, next_player, current_player)
   });
+
+  if (this.state.gameStatus !== Statuses.IN_PROGRESS) {
+    this.highlightWinners(marks);
+  }
+}
+
+highlightWinners = async (marks) => {
+  let data = await PhoenixApi.getWinningIndices(marks);
```
Comment (soft-circles) [inline src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js]: hmmm I don't like that `highlightWinners` both FETCHES the winning indices and it highlights them. Remember the single responsibility principle!

File: src/services/__tests__/phoenixAPI.js [hunk 4]
```diff
@@ -70,3 +70,39 @@ describe("GET status of the game", () => {
...
+    expect(fakeAxios.get).toHaveBeenCalledWith(
+      "https://ttt-json-api.herokuapp.com/winner/",
+      {
+        params: {
+          spots: JSON.stringify(spots).replace(/,/g, ', '),
+        }
+      }
+    );
+  });
+});
```
Comment (soft-circles) [inline src/services/__tests__/phoenixAPI.js:122]: please remove comments

File: src/services/__tests__/phoenixAPI.js [hunk 5]
```diff
@@ -70,3 +70,39 @@ describe("GET status of the game", () => {
...
+      Promise.resolve({
+        data: { index_list: ["1", "2", "3"] }
+      })
+    );
+
+    // work
+    const data = await PhoenixApi.requestWinningIndices(["X", "X", "X", "O", "O", "6", "7", "8", "9"]);
+
+    // expect
+    expect(data.index_list).toEqual(["1", "2", "3"]);
```
Comment (soft-circles) [inline src/services/__tests__/phoenixAPI.js]: if you're going to be expecting the same set of data can we extract it to a variable?
ex:

`const winningIndices = ["1", "2", "3"];`

```
      Promise.resolve({
        data: { index_list: winningIndices }
      })
```

`expect(data.index_list).toEqual(winningIndices);`

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js [hunk 6]
```diff
@@ -51,9 +43,34 @@ class ComputerVsHumanGameComponent extends Component {
...
+
+    highlightWinners = async (marks) => {
+      let data = await PhoenixApi.getWinningIndices(marks);
+      let winningCells = data.index_list;
+      winningCells.forEach((index) => {
+        let indexString = parseInt(index);
+      document.getElementById(indexString).classList.add('highlight');
+      })
```
Comment (soft-circles) [inline src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:74]: again, we are both fetching and highlighting. Let's spit this into smaller methods.

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js (hunk 1 above)
Comment (cmvandrevala) [reply src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:68]: ^ Agree

File: src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js (hunk 3 above)
Comment (cmvandrevala) [reply src/components/humanVsHumanGameComponent/humanVsHumanGameComponent.js]: ^ Agree

File: src/services/__tests__/phoenixAPI.js (hunk 4 above)
Comment (cmvandrevala) [reply src/services/__tests__/phoenixAPI.js:122]: Definitely agree. It is very good that you are doing the four phase test, but you should not need to label the different sections of each test. Whitespace is sufficient

**Review by cmvandrevala — CHANGES_REQUESTED** [review]
I definitely echo what @akiraandy said in his review.

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js (hunk 1 above)
Comment (GrandSuccess87) [reply src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:68]: Okay, absolutely, we've now extracted `updateBoardState` into three separate methods

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js [hunk 7]
```diff
@@ -30,27 +30,50 @@ class ComputerVsHumanGameComponent extends Component {
...
       await this.state.computer.makeMove(board, human);
+
+      await this.updateGameStatus(board.marks(), human, this.state.computer.symbol);
+      this.checkForWinner(board.marks());
+    }
+
+    updateGameStatus = async (marks, next_player, current_player) => {
       this.setState({
-        gameStatus: await PhoenixApi.getStatus(board.marks(), human, this.state.computer.symbol),
+        gameStatus: await this.state.api.getStatus(marks, next_player, current_player)
```
Comment (cmvandrevala) [inline src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:50]: Nice

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js [hunk 8]
```diff
@@ -30,27 +30,50 @@ class ComputerVsHumanGameComponent extends Component {
...
       });
     }
 
+    checkForComputerTurn = () => {
+      if(this.state.gameStatus === Statuses.IN_PROGRESS){
+        this.setState({
+          gameStatus: Statuses.COMPUTER_THINKING
+        });
+        setTimeout(() => {this.computerMove()}, 1000);
```
Comment (cmvandrevala) [inline src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:59]: Why is the `setTimeout` needed here? It is not wrong, per se, just a bit weird to see the computer move behind the timeout.

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.spec.js [hunk 9]
```diff
@@ -1,20 +1,22 @@
...
 function flushPromises() {
   return new Promise(resolve => setImmediate(resolve));
 }
 
-describe('Computer making a move', () => {
+describe('Computer making a move',  () => {
   it('places an X for the first move', () => {
     const computerPlayer = new FakeComputer('X');
-    const wrapper = mount(<ComputerComponent computerPlayer={computerPlayer} />);
+    const wrapper = mount(<ComputerComponent computerPlayer={computerPlayer} api={FakeAPI} />);
```
Comment (cmvandrevala) [inline src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.spec.js:15]: Good use of injection

**Review by cmvandrevala — APPROVED** [review]

File: src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js (hunk 8 above)
Comment (GrandSuccess87) [reply src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js:59]: Initially the computer was placing a marker very quickly.  So we placed `setTimeout` here to extend the time the computer makes a turn and enhance the users' experience.

**Pull request merged**